src/controllers/profileInitialSetupController.js

```diff
@@ -156,20 +170,25 @@ const profileInitialSetupController = function (
     */
   const validateSetupToken = async (req, res) => {
     const { token } = req.body;
-    const currentMoment = moment.tz('America/Los_Angeles');
+    const currentMoment = moment.now();
     try {
       const foundToken = await ProfileInitialSetupToken.findOne({ token });
 
       if (foundToken) {
         const expirationMoment = moment(foundToken.expiration);
 
-        if (expirationMoment.isAfter(currentMoment)) {
-          res.status(200).send(foundToken);
+        if (foundToken.isSetupCompleted) {
+          res.status(400).send(TOKEN_HAS_SETUP_MESSAGE);
+        } else if (foundToken.isCancelled) {
+          res.status(400).send(TOKEN_CANCEL_MESSAGE);
+        } else if (expirationMoment.isBefore(currentMoment)) {
+          res.status(400).send(TOKEN_EXPIRED_MESSAGE);
         } else {
-          res.status(400).send('Invalid token');
-        }
+          res.status(400).send(TOKEN_INVALID_MESSAGE);
```

Comment:
Hi @ptrpengdev, I wanted to bring to your attention that the token will always be invalid. You've set the default condition to res.status(400).send(TOKEN_INVALID_MESSAGE)

Reply:
My mistatke. Updated.